base.js

```diff
@@ -25,9 +35,52 @@ module.exports = {
       },
     ],
 
+    // Require return statements to either always or never specify values
+    'consistent-return': 'error',
+
+    // Require default case in switch statements
+    'default-case': ['error', { commentPattern: '^no default$' }],
+
+    // Enforce default clauses in switch statements to be last
+    'default-case-last': 'error',
+
+    'default-param-last': 'error',
+
+    // Encourage use of dot notation whenever possible
+    'dot-notation': ['error', { allowKeywords: true }],
+
+    // require the use of === and !==
+    eqeqeq: ['error', 'always', { null: 'ignore' }],
+
     // Turn off function paren newline.
     'function-paren-newline': 'off',
 
+    // Require function expressions to have a name
+    'func-names': 'warn',
+
+    // Require all requires be top-level
+    'global-require': 'error',
```

Comment:
note: this rule seems to also have been deprecated (in eslint v7) and scheduled for removal for eslint v11, given that we are progressively migrating to ESM modules anyway this may not that big of a deal and we may just keep it for a little longer and just remove it later on when it is removed or we don't need it anymore, alternatively if we don't care about this that much also in the short term, we may consider dropping it right away.

thought: if we decide to keep this rule in the short term, maybe we could leave a comment to remind us that we have already assessed that we can remove this rule as soon as it is removed or we don't have any project still using the CommonJS module system.